javaslang/src/main/java/javaslang/collection/Vector.java

```diff
@@ -561,20 +566,22 @@ private Vector(int indexShift, HashArrayMappedTrie<Integer, T> trie) {
 
     @Override
     public Vector<T> append(T element) {
-        if (isEmpty()) {
-            return of(element);
-        } else {
-            return new Vector<>(indexShift, trie.put(length() + indexShift, element));
-        }
+        return wrap(trie.appendLeaf(element));
     }
 
     @Override
     public Vector<T> appendAll(Iterable<? extends T> elements) {
-        HashArrayMappedTrie<Integer, T> result = trie;
-        for (T element : elements) {
-            result = result.put(result.size() + indexShift, element);
+        Objects.requireNonNull(elements, "elements is null");
+        if (isEmpty()) {
+            return ofAll(elements);
+        } else {
+            Vector<T> result = this;
+            for (T element : elements) {
+                result = result.append(element);
```

Comment:
Sure, that's how `create` worked previously also: https://github.com/javaslang/javaslang/pull/1449/commits/d0b782f386271cb2fa6c2dcc7d2bcbe2c79a57bc#diff-c6c1876105d111251f38e0916d4d99a4R24

i.e. create the array locally, populate it, put it in the tree and don't keep any other references to it :)